### PR TITLE
[HttpFoundation] Fixed default user-agent (3.X -> 4.X)

### PR DESCRIPTION
**Pull request by lyrixx**
| Q             | A
| ------------- | ---
| Branch?       | 4.0
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #25390
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Review by Simperfit — APPROVED** [review]

**Review by issei-m — APPROVED** [review]

File: src/Symfony/Component/HttpFoundation/Request.php [hunk 1]
```diff
@@ -312,7 +312,7 @@ public static function create($uri, $method = 'GET', $parameters = array(), $coo
             'SERVER_NAME' => 'localhost',
             'SERVER_PORT' => 80,
             'HTTP_HOST' => 'localhost',
-            'HTTP_USER_AGENT' => 'Symfony/3.X',
+            'HTTP_USER_AGENT' => 'Symfony/4.X',
```
Comment (javiereguiluz) [inline src/Symfony/Component/HttpFoundation/Request.php]: ...and the usual question when we do changes like this: can we remove the version number? would it make sense to do that? is it valid as a User Agent string?

Comment (lyrixx) [reply src/Symfony/Component/HttpFoundation/Request.php]: :+1: for removing the version

**Comment by nicolas-grekas** [comment]
Why remove the version number?
For reference, we changed from 2 to 3 without much discussion in #17340
So let's have a real discussion now?
To me, the version seems legit as it provides a hint that could be useful.
I wouldn't mind either removing it, but why in the first place?

**Comment by lyrixx** [comment]
Actually I really don't care. I removed it as @javiereguiluz  suggested it in order to ease maintenance.
More over this value is almost useless, as it's a default value that is almost every-time overrided. I don't really see when this value could give an hint. You may be confused because of the [BrowserKit Client](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/BrowserKit/Client.php#L125) ?

**Review by fabpot — APPROVED** [review]
Removing the version seems the right thing to do.

**Comment by nicolas-grekas** [comment]
Thank you @lyrixx.

**Pull request merged**